File: blog-admin/blog_admin/blog_admin/models/skill.py
from django.db import models


class Skill(models.Model):
    name = models.CharField(max_length=100)
    slug = models.SlugField(max_length=100, unique=True)
    category = models.CharField(max_length=100)
    proficiency = models.IntegerField(default=0)
    icon = models.CharField(max_length=255, blank=True, null=True)
    sort_order = models.IntegerField(default=0)
    is_active = models.BooleanField(default=True)
    created_at = models.DateTimeField(auto_now_add=True)

    class Meta:
        managed = False
        db_table = 'skills'
        ordering = ['sort_order', 'name']

    def __str__(self):
        return f"{self.name} ({self.category})"

File: blog-api/src/test/scala/su/wps/blog/repositories/models/Skill.scala
package su.wps.blog.repositories.models

import cats.syntax.functor.*
import doobie.*
import su.wps.blog.instances.time.*
import su.wps.blog.tools.types.*

import java.time.ZonedDateTime

final case class Skill(
  id: PosInt,
  name: Varchar[W.`100`.T],
  slug: Varchar[W.`100`.T],
  category: Varchar[W.`100`.T],
  proficiency: Int,
  icon: Option[Varchar[W.`255`.T]],
  sortOrder: Int,
  isActive: Boolean,
  createdAt: ZonedDateTime
)

object Skill {

  object sql {
    def insert(skill: Skill): ConnectionIO[Unit] = {
      val sql =
        """
          |INSERT INTO skills (
          | id,
          | name,
          | slug,
          | category,
          | proficiency,
          | icon,
          | sort_order,
          | is_active,
          | created_at
          |) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)
          |""".stripMargin
      Update[Skill](sql).run(skill).void
    }
  }
}
